src/main/java/org/elasticsearch/index/store/Store.java

```diff
@@ -114,6 +119,11 @@ public Store(ShardId shardId, @IndexSettings Settings indexSettings, DirectorySe
         this.directory = new StoreDirectory(directoryService.newFromDistributor(distributor), Loggers.getLogger("index.store.deletes", indexSettings, shardId));
         this.shardLock = shardLock;
         this.onClose = onClose;
+        final TimeValue refreshInterval = indexSettings.getAsTime(INDEX_STORE_STATS_REFRESH_INTERVAL, TimeValue.timeValueSeconds(10));
+        this.statsCache = new StoreStatsCache(refreshInterval);
+        statsCache.getOrRefresh();
```

Comment:
never mind - I see now we always rely on a valid cached object (because of the non blocking nature). In this case I wonder if we should force passing it to the constructor?